src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.ServiceFabric.Client;
+using Microsoft.ServiceFabric.Client.Http;
+using ServiceFabricCommon = Microsoft.ServiceFabric.Common;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class RestHealthCheckPublisher : HealthCheckPublisher
+	{
+		private readonly RestClientWrapper m_clientWrapper;
+
+		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
+
+		protected override string HealthReportSourceIdImpl => nameof(RestHealthCheckPublisher);
+
+		// Taken from https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-environment-variables-reference
+		internal static string FabricNodeNameEnv = "Fabric_NodeName";
+
+		public RestHealthCheckPublisher(ILogger<ServiceFabricHealthCheckPublisher> logger,
+										RestHealthCheckPublisherOptions options,
+										ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
+		{
+			m_clientWrapper = new(new Uri(options.RestHealthPublisherClusterEndpoint));
+			m_logger = logger;
+		}
+
+		public override async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
```

Comment:
It looks to me as this method very similar to the implementation in `ServiceFabricHealthCheckPublisher` and could be also moved to the base class, instead you can create `private abstract Task ReportHealthAsync(SfHealthInformation info)` with implementation diffirences.